packages/server/src/fhir/operations/resourcegraph.ts

```diff
@@ -0,0 +1,344 @@
+import {
+  allOk,
+  badRequest,
+  DEFAULT_SEARCH_COUNT,
+  evalFhirPathTyped,
+  getReferenceString,
+  notFound,
+  OperationOutcomeError,
+  Operator,
+  parseSearchDefinition,
+  PropertyType,
+  toTypedValue,
+  TypedValue,
+} from '@medplum/core';
+import {
+  Bundle,
+  GraphDefinition,
+  GraphDefinitionLink,
+  GraphDefinitionLinkTarget,
+  Reference,
+  Resource,
+} from '@medplum/fhirtypes';
+import { Request, Response } from 'express';
+import { logger } from '../../logger';
+
+import { Repository } from '../repo';
+import { sendResponse } from '../routes';
+
+/**
+ * Handles a Resource $graph request.
+ *
+ * The operation fetches all the data related to this resources as defined by a GraphDefinition resource
+ *
+ * See: https://hl7.org/fhir/plandefinition-operation-apply.html
+ *
+ * @param req The HTTP request.
+ * @param res The HTTP response.
+ */
+export async function resourceGraphHandler(req: Request, res: Response): Promise<void> {
+  const { resourceType, id } = req.params;
+  const repo = res.locals.repo as Repository;
```

Comment:
@reshmakh @rahul1  - This is the key line to make sure "AccessPolicy" is being enforced.  The authentication middleware sets up a `Repository` instance for the user for the duration of the HTTP request.  As long as you use this instance of `repo`, you're good.

Agree with @reshmakh  that it would be a good idea to add a test for future maintainability.